source_gen/lib/src/generator.dart

```diff
@@ -51,11 +51,18 @@ class InvalidGenerationSourceError extends Error {
     final buffer = StringBuffer(message);
 
     if (element != null) {
-      final span = spanForElement(element);
-      buffer
-        ..writeln()
-        ..writeln(span.start.toolString)
-        ..write(span.highlight());
+      try {
+        final span = spanForElement(element);
+        buffer
+          ..writeln()
+          ..writeln(span.start.toolString)
+          ..write(span.highlight());
+      } catch (_) {
+        buffer
+          ..writeln()
+          ..writeln('While logging this error: '
```

Comment:
The wording here is a bit awkward, maybe `Error finding source span information for "$element".`?

Reply:
Yeeeeeah this is a really tough one.

The error message is going to be seen by someone _using_ the generator, who is likely not the generator author. The part of the error message that we already printed contains the main information. Then, the generator author gave us the problem element and we're trying to log that to add more information. I half think it would be better to just log nothing, because the person reading this will likely have no idea what the generator is doing. But I also don't like just completely swallowing the information.

So what I'm aiming for is to give a note that people who don't know the generator will ignore, while still copying+pasting it if they contact the generator author :)

Here's some example output:

```
[SEVERE]: Error running FunkyGenerator
FunkyGenerator failed because you forgot to write "foo bar baz" in your srcs. Sorry. --FunkyGenerator authors.
While logging this error: tried to log source from "class SomeUnavailableDepOfMyClass" but couldn't find it.
```

WDYT?

Thanks :)

Reply:
Maybe just replace `While logging this error` with `Additional Errors` or something?

Reply:
I'm a bit worried they might think the whole problem is caused by SomeUnavailableDepOfMyClass being unavailable, and they'll try to fix the dep on it...when it's actually fine that the src is missing. Maybe less is better:

```
[SEVERE]: Error running FunkyGenerator
FunkyGenerator failed because you forgot to write "foo bar baz" in your srcs. Sorry. --FunkyGenerator authors.
Cause: class SomeUnavailableDepOfMyClass
```

?

Reply:
That seems fine - its basically just falling back on a less descriptive "source span" at that point.

Reply:
SG, done. Thanks!

Reply:
(And added a comment).